kernel: modules: bluetooth: new Bluetooth menu and separating UART and USB drivers

testuser7 opened the pull request:
Not all devices need Bluetooth UART and USB drivers to be installed together. Three separate drivers have been created:
- kmod-bluetooth (core)
- kmod-hci-uart
- kmod-btusb

This PR requires testing on Bluetooth-enabled devices.

testuser7 committed:
kernel: modules: bluetooth: creating a new Bluetooth menu

Move Bluetooth modules to new Bluetooth menu from Other menu.

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/15118
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hauke commented:
Please split this up into two patches. One patch only moving the definitions sand an other doing the changes. Currently it is hard to review the changes.

testuser7 committed:
kernel: modules: bluetooth: separating UART and USB drivers

Not all devices need Bluetooth UART and USB drivers to be installed together. Three separate drivers have been created:
- kmod-bluetooth
- kmod-hci-uart
- kmod-btusb

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/15118
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

testuser7 commented:
> Please split this up into two patches. One patch only moving the definitions sand an other doing the changes. Currently it is hard to review the changes.

Done.

testuser7 committed:
linux: use packages kmod-hci-uart and kmod-btusb instead of kmod-bluetooth

Use Bluetooth UART and USB packages separetly.

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/15118
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hauke commented on package/kernel/linux/modules/bluetooth.mk:
```diff
@@ -0,0 +1,121 @@
+ Kernel support for Bluetooth device with SDIO interface
+endef
+
+$(eval $(call KernelPackage,btsdio))
```
Please add an empty line
